File: experiments/business/openshop/compose/fragment_settings.kt
package com.example.myapplication

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp
import com.example.myapplication.ui.theme.MyApplicationTheme

@Composable

fun fragment_settings() {
    androidx.compose.material3.Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = androidx.compose.foundation.shape.RoundedCornerShape(16.dp),
        color = androidx.compose.material3.MaterialTheme.colorScheme.surface
    ) {
        Column(modifier = Modifier
            .fillMaxWidth()
            .fillMaxHeight()
            .padding(16.0.dp)) {
            Column(modifier = Modifier
                .fillMaxWidth()
                .wrapContentHeight()) {
                Text(
                    text = "Select the country n where you want to shop ",
                    modifier = Modifier
                        .fillMaxWidth()
                        .wrapContentHeight(),
                    color = Color(0xFF212121),
                    fontSize = 17.sp,
                    textAlign = TextAlign.Center
                )
                val spinnerOptions_settings_shop_selection_spinner =
                    listOf("Select", "Option 1", "Option 2", "Option 3")
                var spinnerExpanded_settings_shop_selection_spinner by remember {
                    mutableStateOf(
                        false
                    )
                }
                var spinnerSelected_settings_shop_selection_spinner by remember {
                    mutableStateOf(
                        spinnerOptions_settings_shop_selection_spinner.first()
                    )
                }
                Box(
                    modifier = Modifier
                        .background(Color.Unspecified)
                        .fillMaxWidth()
                        .wrapContentHeight()
                        .padding(top = 8.0.dp)
                ) {
                    OutlinedButton(onClick = {
                        spinnerExpanded_settings_shop_selection_spinner = true
                    }, modifier = Modifier.fillMaxWidth()) {
                        Text(text = spinnerSelected_settings_shop_selection_spinner)
                    }
                    DropdownMenu(
                        expanded = spinnerExpanded_settings_shop_selection_spinner,
                        onDismissRequest = {
                            spinnerExpanded_settings_shop_selection_spinner = false
                        }) {
                        spinnerOptions_settings_shop_selection_spinner.forEach { option ->
                            DropdownMenuItem(
                                text = { Text(option) },
                                onClick = {
                                    spinnerSelected_settings_shop_selection_spinner = option
                                    spinnerExpanded_settings_shop_selection_spinner = false
                                }
                            )
                        }
                    }
                }
            }
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .wrapContentHeight()
                    .padding(top = 16.0.dp)
                    .padding(bottom = 16.0.dp)
            ) {
                Text(
                    text = "Open source libraries",
                    modifier = Modifier
                        .fillMaxWidth()
                        .wrapContentHeight(),
                    color = Color(0xFF212121),
                    fontSize = 17.sp
                )
                Text(
                    text = "License details for open source software",
                    modifier = Modifier
                        .fillMaxWidth()
                        .wrapContentHeight(),
                    color = Color(0xFF727272),
                    fontSize = 15.sp
                )
            }
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .wrapContentHeight()
                    .padding(top = 16.0.dp)
                    .padding(bottom = 16.0.dp)
            ) {
                Text(
                    text = "Privacy policy",
                    modifier = Modifier
                        .fillMaxWidth()
                        .wrapContentHeight(),
                    color = Color(0xFF212121),
                    fontSize = 17.sp
                )
                Text(
                    text = "Basic information about Privacy Policy",
                    modifier = Modifier
                        .fillMaxWidth()
                        .wrapContentHeight(),
                    color = Color(0xFF727272),
                    fontSize = 15.sp
                )
            }

        }
    }
}

@Preview(showBackground = true)
@Composable
fun p14() {
    MyApplicationTheme {
        fragment_settings()
    }
}

File: experiments/business/openshop/swiftui/Fragment_settings.swift
import SwiftUI

struct Fragment_settings: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      VStack(alignment: .leading, spacing: 0) {
        Text("Select the country n where you want to shop ").font(.system(size: 17)).foregroundColor(Color(red: 0.12941176470588237, green: 0.12941176470588237, blue: 0.12941176470588237, opacity: 1)).multilineTextAlignment(.center).frame(maxWidth: .infinity, alignment: .center)
        Picker("", selection: .constant(0)) {
          Text("Select").tag(0)
          Text("Option 1").tag(1)
          Text("Option 2").tag(2)
          Text("Option 3").tag(3)
        }
        .pickerStyle(.menu)
        .background(Image("spinner_background_black_baseline").resizable().scaledToFill())
        .frame(maxWidth: .infinity)
        .padding(.top, 8.0)
      }
      .frame(maxWidth: .infinity)
      VStack(alignment: .leading, spacing: 0) {
        Text("Open source libraries").font(.system(size: 17)).foregroundColor(Color(red: 0.12941176470588237, green: 0.12941176470588237, blue: 0.12941176470588237, opacity: 1)).frame(maxWidth: .infinity, alignment: .leading)
        Text("License details for open source software").font(.system(size: 15)).foregroundColor(Color(red: 0.4470588235294118, green: 0.4470588235294118, blue: 0.4470588235294118, opacity: 1)).frame(maxWidth: .infinity, alignment: .leading)
      }
      .frame(maxWidth: .infinity)
      .padding(.top, 16.0)
      .padding(.bottom, 16.0)
      VStack(alignment: .leading, spacing: 0) {
        Text("Privacy policy").font(.system(size: 17)).foregroundColor(Color(red: 0.12941176470588237, green: 0.12941176470588237, blue: 0.12941176470588237, opacity: 1)).frame(maxWidth: .infinity, alignment: .leading)
        Text("Basic information about Privacy Policy").font(.system(size: 15)).foregroundColor(Color(red: 0.4470588235294118, green: 0.4470588235294118, blue: 0.4470588235294118, opacity: 1)).frame(maxWidth: .infinity, alignment: .leading)
      }
      .frame(maxWidth: .infinity)
      .padding(.top, 16.0)
      .padding(.bottom, 16.0)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .padding(16.0)
  }
}

struct Fragment_settings_Previews: PreviewProvider {
  static var previews: some View {
    Fragment_settings()
  }
}
